src/num/fft.c

```diff
@@ -205,16 +205,53 @@ struct fft_plan_s {
 
 static DEF_TYPEID(fft_plan_s);
 
+static char* fftw_wisdom_name(unsigned int N, bool backwards, unsigned int flags, const long dims[N]) {
+
+	char* tbpath = getenv("TOOLBOX_PATH");
+	if (NULL == tbpath)
+		return NULL;
+	char* loc  = NULL;
+
+	// Space for path and null terminator.
+	size_t space = snprintf(loc , 0, "%s/save/fftw/N_%d_BACKWARD_%d_FLAGS_%d_DIMS", tbpath, N, backwards, flags);
+	// Space for dimensions.
+	for (size_t idx = 0; idx < N; idx ++)
+		space += snprintf(loc, 0, "_%lu", dims[N]);
```

Comment:
this should be dims[idx]. The way I would have implemented it is to have one function which calls snprintf on a given buffer and returns the size if given NULL and does the printing when given a buffer, i.e. instead of duplicating all the printfs doing everything twice just have one overall function which is called twice - one time to get the size and then to actually print.

Reply:
Sorry, that was stupid of me again to use `N` instead of `idx`. I'll make that change right now.
On avoiding using multiple `printf`s, I wanted to avoid having multiple buffers in order to add the `dims` values sequentially so that the string varied with `N` as well and using `strcat` with smaller space calculations seemed the easiest to me! I'm not attached to this implementation however, and if you'd like, I can change it.